### PR TITLE
Add a phone number column to the user_accounts table

### PR DESCRIPTION
**Pull request by plunkettgoogle**
Add a phone number column to the user_accounts table.

## Description

This change adds a phone number to the user_account table, adds the corresponding migration, and updates the swagger API.

## How Can This Be Tested/Reviewed?

0. Start a local server via `yarn dev:all`.
1. Go to http://localhost:3100/docs and use the auth/login endpoint to log in as one of the test accounts.
2. Use the auth token to set the bearer token for subsequent requests (click the lil' "lock" icon, and paste the auth token)
3. GET the user JSON through the user endpoint
4. Paste the user JSON into the request body of the PUT user/{id} endpoint, and modify the phone number field
5. GET the user JSON again through the user endpoint, and verify the phone number was set.

To run the added test:
`cd backend/core && yarn test:e2e:local --testPathPattern user.e2e-spec`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [x] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by plunkettgoogle** [comment]
Hi @seanmalbert, Gabe mentioned you might be interested in this PR that introduces phone numbers to the user_accounts table. We've merged this already in CityOfDetroit. I'm not sure about the migration now though, since I created the migration before a few of the ones that exist here (e.g. https://github.com/bloom-housing/bloom/blob/dev/backend/core/src/migration/1634912142711-remove-preferences-page.ts). Do migrations run in-order based on the timestamp?

File: backend/core/src/auth/entities/user.entity.ts [hunk 1]
```diff
@@ -77,6 +86,11 @@ export class User {
   @Type(() => Date)
   dob?: Date | null
 
+  @Column("varchar", { nullable: true })
+  @Expose()
+  @IsPhoneNumber(null)
```
Comment (pbn4) [inline backend/core/src/auth/entities/user.entity.ts:91]: This validator is missing groups parameter, please see other validators.

**Review by pbn4 — CHANGES_REQUESTED** [review]

**Comment by pbn4** [comment]
> I'm not sure about the migration now though, since I created the migration before a few of the ones that exist here (e.g. https://github.com/bloom-housing/bloom/blob/dev/backend/core/src/migration/1634912142711-remove-preferences-page.ts). Do migrations run in-order based on the timestamp?

Yes in case many migrations are to be applied they will be run in order based on timestamp, if there is a migration with more recent timestamp already applied the older one will still get applied on next migration:run too (typeorm will figure out it's missing even though it has an older timestamp).

File: backend/core/src/auth/entities/user.entity.ts (hunk 1 above)
Comment (plunkettgoogle) [reply backend/core/src/auth/entities/user.entity.ts]: Fixed.

**Comment by plunkettgoogle** [comment]
@pbn4, does this look OK to you now? The test failure seems to be unrelated to the change.

**Review by seanmalbert — APPROVED** [review]
This looks good, @plunkettgoogle! Thank you for contributing this addition!

**Pull request merged**